File: examples/src/main/kotlin/usageExamples/distinct/Distinct.kt
package usageExamples.distinct

// :replace-start: {
//    "terms": {
//       "CONNECTION_URI_PLACEHOLDER": "\"<connection string uri>\"",
//       "import io.github.cdimascio.dotenv.dotenv\n": ""
//    }
// }
// :snippet-start: distinct-usage-example
import com.mongodb.MongoException
import com.mongodb.client.model.Filters
import com.mongodb.kotlin.client.coroutine.MongoClient
import io.github.cdimascio.dotenv.dotenv
import kotlinx.coroutines.runBlocking

data class Movie(val year: Int, val directors: List<String>)

fun main() = runBlocking {
    // :remove-start:
    val dotenv = dotenv()
    val CONNECTION_URI_PLACEHOLDER = dotenv["MONGODB_CONNECTION_URI"]
    // :remove-end:
    // Replace the uri string with your MongoDB deployment's connection string
    val uri = CONNECTION_URI_PLACEHOLDER
    val mongoClient = MongoClient.create(uri)
    val database = mongoClient.getDatabase("sample_mflix")
    val collection = database.getCollection<Movie>("movies")

    collection.insertMany(listOf(Movie(1992,listOf("Carl Franklin")), Movie(1995,listOf("Carl Franklin")), Movie(1998,listOf("Carl Franklin"))))// :remove:
    try {
        val resultsFlow = collection.distinct<Int>(
            Movie::year.name, Filters.eq(Movie::directors.name, "Carl Franklin")
        )
        resultsFlow.collect { println(it) }
    } catch (e: MongoException) {
        System.err.println("An error occurred: $e")
    }

    // :remove-start:
    // clean up
    database.drop()
    // :remove-end:
    mongoClient.close()
}
// :snippet-end:
// :replace-end:

File: content/node/v6.x/source/code-snippets/usage-examples/distinct.js
import { MongoClient } from "mongodb";

// Replace the uri string with your MongoDB deployment's connection string.
const uri = "<connection string uri>";

const client = new MongoClient(uri);

async function run() {
  try {
    
    // Get the database and collection on which to run the operation
    const database = client.db("sample_mflix");
    const movies = database.collection("movies");

    // Specify the document field to find distinct values for
    const fieldName = "year";

    // Specify an optional query document to narrow results
    const query = { directors: "Barbra Streisand" };

    // Execute the distinct operation
    const distinctValues = await movies.distinct(fieldName, query);

    // Print the result
    console.log(distinctValues);
  } finally {
    await client.close();
  }
}
run().catch(console.dir);
